File: app/src/main/java/com/itba/runningMate/repository/runningstate/RunningStateStorage.java
package com.itba.runningMate.repository.runningstate;

public interface RunningStateStorage {

    String LANDING_STATE_PREFERENCES_FILE = "PREF_RUNNING_MATE_RUNNING_STATE";

    boolean isCenterCamera();

    void setCenterCamera(final boolean centerCamera);

    double getLastKnownLatitude();

    double getLastKnownLongitude();

    boolean hasLastKnownLocation();

    void setLastKnownLocation(final double latitude, final double longitude);

    void persistState();
}

File: app/src/main/java/com/itba/runningMate/repository/runningstate/RunningStateStorage.kt
package com.itba.runningMate.repository.runningstate

interface RunningStateStorage {

    fun isCenterCamera(): Boolean

    fun setCenterCamera(centerCamera: Boolean)

    fun getLastKnownLatitude(): Double

    fun getLastKnownLongitude(): Double

    fun hasLastKnownLocation(): Boolean

    fun setLastKnownLocation(latitude: Double, longitude: Double)

    fun persistState()

    companion object {
        const val LANDING_STATE_PREFERENCES_FILE = "PREF_RUNNING_MATE_RUNNING_STATE"
    }
}